src/authorizations/components/redesigned/ResourceAccordion.tsx

```diff
@@ -0,0 +1,154 @@
+// Libraries
+import React, {Component} from 'react'
+
+// Clockface
+import {Accordion} from '@influxdata/clockface'
+import {ResourceAccordionHeader} from './ResourceAccordionHeader'
+import GetResources from 'src/resources/components/GetResources'
+import {ResourceAccordionBody} from './ResourceAccordionBody'
+
+// Types
+import {AppState, Telegraf, ResourceType} from 'src/types'
+
+// Selectors
+import {getAll} from 'src/resources/selectors'
+import {connect} from 'react-redux'
+import {Bucket} from 'src/client'
+
+export type ReadWritePermissions = {
+  read: boolean
+  write: boolean
+}
+export type SublevelPermissions = {
+  [x: string]: ReadWritePermissions
+}
+
+interface OwnProps {
+  resources: string[]
+}
+interface State {
+  permissions: any
+}
+interface StateProps {
+  telegrafPermissions: any
+  bucketPermissions: any
+}
+
+type Props = OwnProps & StateProps
+
+class ResourceAccordion extends Component<Props, State> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      permissions: {
+        telegrafs: props.telegrafPermissions,
+        buckets: props.bucketPermissions,
+      },
+    }
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    if (!props.telegrafPermissions) {
+      return null
+    }
+    return {
+      ...state,
+      permissions: {
+        telegrafs: props.telegrafPermissions,
+        buckets: props.bucketPermissions,
+      },
+    }
+  }
+
+  public render() {
+    const {resources} = this.props
+    const {permissions} = this.state
+
+    if (resources) {
```

Comment:
stylistic nit: it's probably simpler and easier to follow if we invert this and do

```ts
if (!resources) {
  return null
}
```
and then have the `resources.map` be the default codepath

Reply:
good one :)